common/pkg/capi/clustertopology/handlers/mutation/meta_test.go

```diff
@@ -37,6 +37,7 @@ func (h *testHandler) Mutate(
 	_ map[string]apiextensionsv1.JSON,
 	holderRef runtimehooksv1.HolderReference,
 	_ client.ObjectKey,
+	_ ClusterGetter,
```

Comment:
To me this really smells like an interface, something like:

```
type ClusterInfoGetter interface {
  func Cluster(context.Context) (*clusterv1.Cluster, error)
  func InfrastructureRefKind(context.Context) string
}
```

WDYT?

Reply:
I think it would simplify writing unit tests too.

Reply:
an interface would make writing unit tests easier. however this won't be that straight forward.

every single unit test creates a NewMetaGeneratePatchesHandler for its test. 

```golang
    return mutation.NewMetaGeneratePatchesHandler("", cl, NewPatch()).(mutation.GeneratePatches)
```

to simplify the tests to allow mocks we would need to change the function definition to take in a constructor to create this interface

changing this to (not sure about this `Factory` name

```golang

type ClusterInfoGetterFactory interface {
  CreateClusterInfoGetter() (ClusterInfoGetter, error)
}

type metaGeneratePatches struct {
	name     string
	mutators []MetaMutator
	clusterInfoGetter ClusterInfoGetter
}

func NewMetaGeneratePatchesHandler(
	name string,
	factory ClusterInfoGetterFactory,
	mutators ...MetaMutator,
) (handlers.Named, error) {
        getter, err := factory.CreateClusterInfoGetter()
        if err != nil {
           return nil, err
        }
	return metaGeneratePatches{
		name:     name,
		mutators: mutators,
		clusterInfoGetter: getter,
	}, nil
}
```
the metaMutator interface might change as follows

```golang
type MetaMutator interface {
	Mutate(
		ctx context.Context,
		obj *unstructured.Unstructured,
		vars map[string]apiextensionsv1.JSON,
		holderRef runtimehooksv1.HolderReference,
               clusterGetter ClusterInfoGetter,
	) error
}
```

I think this would allow us to pass in our implementations of CreateClusterInfoGetters and also allow us to have a default one from client.

With all this being said, I don't really know if all this effort is going to be worth it.